components/org.wso2.carbon.identity.query.saml/src/test/java/org/wso2/carbon/identity/query/saml/X509CredentialImplTest.java

```diff
@@ -0,0 +1,124 @@
+ */
+package org.wso2.carbon.identity.query.saml;
+
+import org.testng.annotations.Test;
+
+import java.math.BigInteger;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.PublicKey;
+import java.security.cert.CertificateParsingException;
+import java.security.cert.X509Certificate;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.RSAPublicKeySpec;
+import java.util.ArrayList;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * Test Class for the X509CredentialImpl
+ */
+public class X509CredentialImplTest {
+    String modulusString = "00d56047acf652298e3fcdbb8cecbc32214722aa1625f88480cf570cee373ada932b140c29b00dc" +
+            "44f6e59e7018dddca66b2f1c645dacb9d4a45459cfa8f7e33df";
+    String exponentString = "18bc01730656bde47476f7cfbd3d8f9e15ede9c389814672dc161e349b08627fc885fe9d2442ae92" +
+            "f0214c7e97cf0b9a9fc876df4f53517ab63d710f997b2779";
+
+    BigInteger modulus = new BigInteger(modulusString, 16);
+    BigInteger exponent = new BigInteger(exponentString, 16);
+
+    X509CredentialImpl testclass = new X509CredentialImpl(modulus, exponent);
+
+    public X509CredentialImplTest() throws InvalidKeySpecException, NoSuchAlgorithmException, CertificateParsingException {
+    }
+
+    @Test
+    public void testGetEntityId() {
+
+        assertEquals(testclass.getEntityId(), null);
+    }
+
+    @Test
+    public void testGetUsageType() {
+
+        assertEquals(testclass.getUsageType(), null);
+    }
+
+    @Test
+    public void testGetKeyNames() {
+
+        assertEquals(testclass.getKeyNames(), null);
+    }
+
+    @Test
+    public void testGetPublicKey() throws InvalidKeySpecException, NoSuchAlgorithmException {
+
+        RSAPublicKeySpec spec = new RSAPublicKeySpec(modulus, exponent);
+        KeyFactory keyFactory = KeyFactory.getInstance("RSA");
+        PublicKey publicKey = keyFactory.generatePublic(spec);
+        assertEquals(testclass.getPublicKey(), publicKey);
+    }
+
+    @Test
+    public void testGetPrivateKey() {
+
+        assertEquals(testclass.getPrivateKey(), null);
+    }
+
+    @Test
+    public void testGetSecretKey() {
+
+        assertEquals(testclass.getSecretKey(), null);
+    }
+
+    @Test
+    public void testGetCredentialContextSet() {
+
+        assertEquals(testclass.getCredentialContextSet(), null);
+    }
+
+    @Test
+    public void testGetCredentialType() {
+
+        assertEquals(testclass.getCredentialType(), null);
+    }
+
+    @Test
+    public void testGetEntityCertificateChain() {
+
+        assertTrue(testequalarray((ArrayList) testclass.getEntityCertificateChain(), new ArrayList<X509Certificate>()));
+    }
+
+    @Test
+    public void testGetCRLs() {
+        assertEquals(testclass.getCRLs(), null);
+    }
+
+    private boolean testequalarray(ArrayList actual, ArrayList expected) {
+
+        for (int i = 0; i < actual.size(); i++) {
```

Comment:
This is a very old school way of writing for loops. Can you write a java8 lambda for this?

Reply:
okay